tests/unit/Libraries/Archive/ZipTest.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace Quantum\Tests\Libraries\Archive {
+
+    use Quantum\Libraries\Storage\FileSystem;
+    use Quantum\Libraries\Archive\Zip;
+    use Quantum\Tests\AppTestCase;
+
+    class ZipTest extends AppTestCase
+    {
+
+        private $fs;
+        private $filename;
+
+        public function setUp(): void
+        {
+            parent::setUp();
+            $this->fs = new FileSystem();
+            $this->filename = base_dir() . DS . 'test.zip';
+        }
+
+        public function tearDown(): void
+        {
+            if ($this->fs->exists($this->filename)) {
+                $this->fs->remove($this->filename);
+            }
+        }
+
+        public function testAddEmptyDirToZip()
+        {
+            $newZip = new Zip($this->filename);
```

Comment:
Any particular reason to not instantiate `new Zip` in the `setUp()` method ?

Reply:
Create `Adapters` directory inside Archive library and move the `PharArchive.php` and `Zip.php` (rename to `ZipArchive.php`) into it.